include/experimental/__p1673_bits/blas1_dot.hpp

```diff
@@ -43,6 +43,7 @@
 #ifndef LINALG_INCLUDE_EXPERIMENTAL___P1673_BITS_BLAS1_DOT_HPP_
 #define LINALG_INCLUDE_EXPERIMENTAL___P1673_BITS_BLAS1_DOT_HPP_
 
+#include <ranges>
```

Comment:
`<ranges>` is a C++20 include.  Would you consider protecting both the include and the use of `iota_view` below with the appropriate feature test macro, and providing a fall-back implementation?  It's OK if the fall-back is not parallel.

Reply:
Sure! Can you point me to an example of using a feature test macro?

Reply:
Sure!  [This web page](https://en.cppreference.com/w/cpp/utility/feature_test) gives a good summary.

```c++
// Ensure that the feature test macro __cpp_lib_ranges is available;
// <version> also defines this macro, but that is a C++20 header.
#include <algorithm>

#if defined(__cpp_lib_ranges)
#  include <ranges>
#endif

void some_function() {
#if defined(__cpp_lib_ranges_iota)
  // ... code using views::iota ...
#else
  // ... fall-back code ...
#endif
}
```

The point of using two different macros -- one for the header, and one for the specific feature `iota::view` -- is that the feature came after the header, so some compiler versions may have the header but not the feature.